src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertTo-Html.cs

```diff
@@ -647,6 +669,40 @@ private void WritePropertyValue(StringBuilder Listtag, MshParameter p)
             }
         }
 
+        /// <summary>
+        /// Check if a string or substrings in a string is uri, if true, then change to hyperlink format.
+        /// </summary>
+        /// <param name="s"> the string that needs to be altered </param>
+        /// <param name="hyper"> if true, then change to hyperlink format </param>
+        /// <returns>the midified string/unmodified string.</returns>
+        private static string CheckUri(string s, bool hyper)
+        {
+            if (hyper)
+            {
+                Regex regHttp = new Regex(@"(?<=\()\b(https?://|www\.)[-A-Za-z0-9+&@#/%?=~_()|!:,.;]*[-A-Za-z0-9+&@#/%=~_()|](?=\))|(?<=(?<wrap>[=~|_#]))\b(https?://|www\.)&[-A-Za-z0-9+&@#/%?=~_()|!:,.;]*[-A-Za-z0-9+&@#/%=~_()|](?=\k<wrap>)|\b(https?://|www\.)[-A-Za-z0-9+&@#/%?=~_()|!:,.;]*[-A-Za-z0-9+&@#/%=~_()|]", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+                var p = "[[[replace:period]]]";
+                s = Regex.Replace(s, @"(?<=\d)\.(?=\d)", p);
+                var htmls = regHttp.Matches(s);
+                for (int i = 0; i < htmls.Count; i++)
+                {
+                    var temp = htmls[i].ToString();
```

Comment:
one thing you could do is try to construct a `uri` - there's a property `scheme` on it that can be used to determine whether http or https was used

Reply:
I think using System.Uri may be better here as it does other validation that the URI is valid.